sys/net/include/ieee802154_frame.h

```diff
@@ -34,19 +36,53 @@ extern "C" {
 /* ...and FCS*/
 #define IEEE_802154_FCS_LEN             (2)
 
-typedef enum __attribute__((packed)) {
-    IEEE_802154_BEACON_FRAME    = 0,
-    IEEE_802154_DATA_FRAME      = 1,
-    IEEE_802154_ACK_FRAME       = 2,
-    IEEE_802154_MAC_CMD_FRAME   = 3
-} ieee802154_frame_type_t;
-
-#define IEEE_802154_SHORT_ADDR_M        (2)
-#define IEEE_802154_LONG_ADDR_M         (3)
+/**
+ * @brief   IEEE802.15.4 FCF bitfield definitions
+ * @{
+ */
+#define IEEE_802154_FCF_TYPE_MASK       (0x0700)
+#define IEEE_802154_FCF_TYPE_POS        (8U)
+#define IEEE_802154_FCF_TYPE_BEACON     (0x0000)
+#define IEEE_802154_FCF_TYPE_DATA       (0x0100)
+#define IEEE_802154_FCF_TYPE_ACK        (0x0200)
+#define IEEE_802154_FCF_TYPE_MAC_CMD    (0x0300)
+
+#define IEEE_802154_FCF_SEC_EN          (0x0800)
+#define IEEE_802154_FCF_SEC_EN_POS      (11U)
+
+#define IEEE_802154_FCF_FRAME_PEND      (0x1000)
+#define IEEE_802154_FCF_FRAME_PEND_POS  (12U)
+
+#define IEEE_802154_FCF_ACK_REQ         (0x2000)
+#define IEEE_802154_FCF_ACK_REQ_POS     (13U)
+
+#define IEEE_802154_FCF_PANID_COMP      (0x4000)
+#define IEEE_802154_FCF_PANID_COMP_POS  (14U)
+
+#define IEEE_802154_FCF_DST_ADDR_MASK   (0x000a)
+#define IEEE_802154_FCF_DST_ADDR_POS    (2U)
+#define IEEE_802154_FCF_DST_ADDR_EMPTY  (0x0000)
+#define IEEE_802154_FCF_DST_ADDR_SHORT  (0x0008)
+#define IEEE_802154_FCF_DST_ADDR_LONG   (0x000a)
+
+#define IEEE_802154_FCF_FRAME_VER_MASK  (0x0030)
+#define IEEE_802154_FCF_FRAME_VER_POS   (4U)
+
+#define IEEE_802154_FCF_SRC_ADDR_MASK   (0x00a0)
+#define IEEE_802154_FCF_SRC_ADDR_POS    (6U)
+#define IEEE_802154_FCF_SRC_ADDR_EMPTY  (0x0000)
+#define IEEE_802154_FCF_SRC_ADDR_SHORT  (0x0080)
+#define IEEE_802154_FCF_SRC_ADDR_LONG   (0x00a0)
+/** @} */
 
+/**
+ * @brief IEEE802.15.4 multi-cast addresses
+ * @{
+ */
 #define IEEE_802154_SHORT_MCAST_ADDR    (0xffff)
-#define IEEE_802154_LONG_MCAST_ADDR     {{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, \
-                                          0xff, 0xff}}
+#define IEEE_802154_LONG_MCAST_ADDR     {{0xff, 0xff, 0xff, 0xff, \
```

Comment:
Any idea, why there are double braces here? If they would be in line with the addresses in `ieee802154_frame_t` they should be single braces (and `IEEE_802154_SHORT_MCAST_ADDR` should be an array, too). Maybe @OlegHahm or the original devs can shed some light upon this.